drivers/java/src/main/java/com/rethinkdb/net/Connection.java

```diff
@@ -345,7 +345,7 @@ public void runNoReply(ReqlAst term, OptArgs globalOpts) {
 
     void stop(Cursor cursor) {
         // The server does reply to a stop request, though the response doesn't have a value.
-        runQuery(Query.stop(cursor.token));
+        runQueryNoreply(Query.stop(cursor.token));
```

Comment:
Yuuuuppp. `runQuery` in `stop()` would cause the memory leak to come back. Nice hunting. :+1: 

IIRC, @danielmewes made the change in:
https://github.com/rethinkdb/rethinkdb/commit/91511142dd55328cfae120fc7500ba3e0e1b402f#diff-f233fec96ca1256a7793f216a89cb657R348

In reference to: https://github.com/rethinkdb/rethinkdb/issues/5938#ref-commit-9151114 ... da code be buggin' yo.

-----

:fire: :volcano: [***"We're building it up, to break it back down..."***](https://www.youtube.com/watch?v=dxytyRy-O1k)